pilot/pkg/model/validation.go

```diff
@@ -1529,6 +1566,42 @@ func ValidateAuthorizationPolicy(name, namespace string, msg proto.Message) erro
 	return nil
 }
 
+// isFirstClassFieldEmpty return false if there is at least one first class field (e.g. properties)
+func isFirstClassFieldEmpty(subject *rbac.Subject) bool {
+	return len(subject.User) == 0 && len(subject.Group) == 0 && len(subject.Properties) == 0 &&
+		len(subject.Namespaces) == 0 && len(subject.NotNamespaces) == 0 && len(subject.Groups) == 0 &&
+		len(subject.NotGroups) == 0 && len(subject.Ips) == 0 && len(subject.NotIps) == 0 &&
+		len(subject.Names) == 0 && len(subject.NotNames) == 0
+}
+
+// Returns true if the user defines a property that already exists in the first-class fields, false
+// if none has overlapped.
+// In the future when we introduce expression conditions in properties field, we might need to revisit
+// this function.
+func hasExistingFirstClassFieldInBinding(propertiesKey string, subject *rbac.Subject) bool {
+	switch propertiesKey {
+	case "source.principal":
```

Comment:
nit. Can we define all these constants for the first class fields?

Reply:
Could you be more specific? Just to clarify, "source.principal" is not a first class field, but subject.Names is.